main_loop_tf/main.py

```diff
@@ -258,11 +259,30 @@ def __run(build_model):
     with tf.Graph().as_default() as graph:
         cfg.global_step = tf.Variable(0, trainable=False, name='global_step',
                                       dtype='int32')
-        inputs = tf.placeholder(shape=cfg.input_shape,
-                                dtype=cfg._FLOATX, name='inputs')
-        val_inputs = tf.placeholder(shape=cfg.val_input_shape,
-                                    dtype=cfg._FLOATX, name='val_inputs')
-        labels = tf.placeholder(shape=[None], dtype='int32', name='labels')
+
+        # Create input placeholders for each gpu tower
+        # All the placeholders are instantiated with a default value
+        # since tensorflow result on a fault if one placeholder is not
```

Comment:
since we want to be able to use a subset of them when the batchsize is not big enough to fill all the GPUs and TF raises a negative shape error if one placeholder is not fed (and has no default value). This is not a problem since when the default values is being used we ignore it's output and do not perform gradient descent on it.